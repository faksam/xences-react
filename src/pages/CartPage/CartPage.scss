.CartPage {
  .productImgCol {
    height: 8rem;

    img {
      width: 100%;
    }
  }

  .productRow {
    margin: 0;
    // white-space: nowrap;
  }

  .productCount {
    margin: 0.25rem;
    float: right;
  }

  .productPrice {
    padding: 1rem;
    font-weight: bold;
    float: right;
  }

  .productCo {
    float: right;
  }

  .productColor {
    padding-top: 1rem;
  }

  .productDetails {
    // padding: 0.8rem;
    overflow-wrap: break-word;
  }

  .productDeleteBtn {
    padding: 1rem;
  }

  .productName {
    overflow-wrap: break-word;
    word-wrap: break-word;
    hyphens: auto;
  }
}
