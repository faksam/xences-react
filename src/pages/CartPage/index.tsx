// import libraries
import * as React from 'react';

// third-party libraries
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

// styles
import './CartPage.scss';

// components
import {
  Button,
  Card,
  Col,
  Icon,
  Row,
  Select,
  TextInput,
} from 'react-materialize';
import ProductCard from '../../components/ProductCard';

// interfaces

// helper functions

// fixtures
import { products } from '../../store/modules/products/fixtures';

export class CartPage extends React.Component<any, any> {
  constructor(props) {
    super(props);

    const { match } = props;
  }

  render() {
    const cartProducts = products.slice(0, 3);
    const subTotal = '6000.00';
    return (
      <React.Fragment>
        <div className="CartPage">
          <Row>
            <Col s={8}>
              {cartProducts.reverse().map(product => (
                <Row key={product.id} className="productRow">
                  <Card className="card">
                    <Row>
                      <Col s={3} className="productImgCol"><img src={product.pictures} /></Col>
                      <Col s={9}>
                          <Col s={9}>
                            <p>{product.name}</p>
                            <p className="productName">{product.description}</p>
                            <p>Red</p>
                            <p>&#8358;{product.price}</p>
                          </Col>
                          <Col s={3} className="productCo">
                            <TextInput label="Quantity" placeholder="1" type="number" max="100" min="1" />
                          </Col>
                      </Col>
                    </Row>
                  </Card>
                </Row>
              ))}
            </Col>
            <Col s={4}>
            <Card title="Order Summary">
              <Row>
              <hr />
              <p>{`SUBTOTAL: ${subTotal}`}</p>
              <p>{`SHIPPING: 500.00`}</p>
              <hr />
              <p><b>GRAND TOTAL: 6500.00</b></p>
              </Row>

              <Link className="btn" to="/checkout">GO TO CHECKOUT</Link>
            </Card>
            </Col>
          </Row>
        </div>
      </React.Fragment>
    );
  }
}

export default CartPage;
