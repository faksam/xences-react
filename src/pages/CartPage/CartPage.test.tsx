// react libraries
import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

// third-party libraries
import { mount } from 'enzyme';

// components
import CartPage from './';

describe('The CartPage component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Router>
        <CartPage />
      </Router>
    );
  });

  it('should be render products page', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
