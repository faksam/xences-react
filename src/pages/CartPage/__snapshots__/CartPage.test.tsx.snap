// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CartPage component should be render products page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CartPage>
      <div
        className="CartPage"
      >
        <Row>
          <div
            className="row"
          >
            <Col
              s={8}
            >
              <div
                className="col s8"
              >
                <Row
                  className="productRow"
                  key="3"
                >
                  <div
                    className="row productRow"
                  >
                    <Card
                      className="card"
                    >
                      <div
                        className="card card"
                      >
                        <div
                          className="card-content"
                        >
                          <div>
                            <Row>
                              <div
                                className="row"
                              >
                                <Col
                                  className="productImgCol"
                                  s={3}
                                >
                                  <div
                                    className="col s3 productImgCol"
                                  >
                                    <img
                                      src="https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg"
                                    />
                                  </div>
                                </Col>
                                <Col
                                  s={9}
                                >
                                  <div
                                    className="col s9"
                                  >
                                    <Col
                                      s={9}
                                    >
                                      <div
                                        className="col s9"
                                      >
                                        <p>
                                          Nice coat
                                        </p>
                                        <p
                                          className="productName"
                                        >
                                          A beautiful nice coat designed by the freshest designer.
                                        </p>
                                        <p>
                                          Red
                                        </p>
                                        <p>
                                          ₦
                                          2000
                                        </p>
                                      </div>
                                    </Col>
                                    <Col
                                      className="productCo"
                                      s={3}
                                    >
                                      <div
                                        className="col s3 productCo"
                                      >
                                        <TextInput
                                          label="Quantity"
                                          max="100"
                                          min="1"
                                          placeholder="1"
                                          type="number"
                                        >
                                          <div
                                            className="input-field col"
                                          >
                                            <input
                                              className=""
                                              id={2}
                                              max="100"
                                              min="1"
                                              placeholder="1"
                                              type="number"
                                            />
                                            <label
                                              className="active"
                                              htmlFor={2}
                                            >
                                              Quantity
                                            </label>
                                            <span
                                              className="helper-text"
                                            />
                                          </div>
                                        </TextInput>
                                      </div>
                                    </Col>
                                  </div>
                                </Col>
                              </div>
                            </Row>
                          </div>
                        </div>
                        <div
                          className="card-reveal"
                        >
                          <span
                            className="card-title"
                          >
                            <Icon
                              right={true}
                            >
                              <i
                                className="material-icons right"
                              >
                                close
                              </i>
                            </Icon>
                          </span>
                        </div>
                      </div>
                    </Card>
                  </div>
                </Row>
                <Row
                  className="productRow"
                  key="2"
                >
                  <div
                    className="row productRow"
                  >
                    <Card
                      className="card"
                    >
                      <div
                        className="card card"
                      >
                        <div
                          className="card-content"
                        >
                          <div>
                            <Row>
                              <div
                                className="row"
                              >
                                <Col
                                  className="productImgCol"
                                  s={3}
                                >
                                  <div
                                    className="col s3 productImgCol"
                                  >
                                    <img
                                      src="https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg"
                                    />
                                  </div>
                                </Col>
                                <Col
                                  s={9}
                                >
                                  <div
                                    className="col s9"
                                  >
                                    <Col
                                      s={9}
                                    >
                                      <div
                                        className="col s9"
                                      >
                                        <p>
                                          Nice cap
                                        </p>
                                        <p
                                          className="productName"
                                        >
                                          A beautiful nice cap designed by the freshest designer.
                                        </p>
                                        <p>
                                          Red
                                        </p>
                                        <p>
                                          ₦
                                          2000
                                        </p>
                                      </div>
                                    </Col>
                                    <Col
                                      className="productCo"
                                      s={3}
                                    >
                                      <div
                                        className="col s3 productCo"
                                      >
                                        <TextInput
                                          label="Quantity"
                                          max="100"
                                          min="1"
                                          placeholder="1"
                                          type="number"
                                        >
                                          <div
                                            className="input-field col"
                                          >
                                            <input
                                              className=""
                                              id={3}
                                              max="100"
                                              min="1"
                                              placeholder="1"
                                              type="number"
                                            />
                                            <label
                                              className="active"
                                              htmlFor={3}
                                            >
                                              Quantity
                                            </label>
                                            <span
                                              className="helper-text"
                                            />
                                          </div>
                                        </TextInput>
                                      </div>
                                    </Col>
                                  </div>
                                </Col>
                              </div>
                            </Row>
                          </div>
                        </div>
                        <div
                          className="card-reveal"
                        >
                          <span
                            className="card-title"
                          >
                            <Icon
                              right={true}
                            >
                              <i
                                className="material-icons right"
                              >
                                close
                              </i>
                            </Icon>
                          </span>
                        </div>
                      </div>
                    </Card>
                  </div>
                </Row>
                <Row
                  className="productRow"
                  key="1"
                >
                  <div
                    className="row productRow"
                  >
                    <Card
                      className="card"
                    >
                      <div
                        className="card card"
                      >
                        <div
                          className="card-content"
                        >
                          <div>
                            <Row>
                              <div
                                className="row"
                              >
                                <Col
                                  className="productImgCol"
                                  s={3}
                                >
                                  <div
                                    className="col s3 productImgCol"
                                  >
                                    <img
                                      src="https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg"
                                    />
                                  </div>
                                </Col>
                                <Col
                                  s={9}
                                >
                                  <div
                                    className="col s9"
                                  >
                                    <Col
                                      s={9}
                                    >
                                      <div
                                        className="col s9"
                                      >
                                        <p>
                                          Nice shirt
                                        </p>
                                        <p
                                          className="productName"
                                        >
                                          A beautiful nice shirt designed by the freshest designer.
                                        </p>
                                        <p>
                                          Red
                                        </p>
                                        <p>
                                          ₦
                                          2000
                                        </p>
                                      </div>
                                    </Col>
                                    <Col
                                      className="productCo"
                                      s={3}
                                    >
                                      <div
                                        className="col s3 productCo"
                                      >
                                        <TextInput
                                          label="Quantity"
                                          max="100"
                                          min="1"
                                          placeholder="1"
                                          type="number"
                                        >
                                          <div
                                            className="input-field col"
                                          >
                                            <input
                                              className=""
                                              id={4}
                                              max="100"
                                              min="1"
                                              placeholder="1"
                                              type="number"
                                            />
                                            <label
                                              className="active"
                                              htmlFor={4}
                                            >
                                              Quantity
                                            </label>
                                            <span
                                              className="helper-text"
                                            />
                                          </div>
                                        </TextInput>
                                      </div>
                                    </Col>
                                  </div>
                                </Col>
                              </div>
                            </Row>
                          </div>
                        </div>
                        <div
                          className="card-reveal"
                        >
                          <span
                            className="card-title"
                          >
                            <Icon
                              right={true}
                            >
                              <i
                                className="material-icons right"
                              >
                                close
                              </i>
                            </Icon>
                          </span>
                        </div>
                      </div>
                    </Card>
                  </div>
                </Row>
              </div>
            </Col>
            <Col
              s={4}
            >
              <div
                className="col s4"
              >
                <Card
                  title="Order Summary"
                >
                  <div
                    className="card"
                  >
                    <div
                      className="card-content"
                    >
                      <span
                        className="card-title"
                      >
                        Order Summary
                      </span>
                      <div>
                        <Row>
                          <div
                            className="row"
                          >
                            <hr />
                            <p>
                              SUBTOTAL: 6000.00
                            </p>
                            <p>
                              SHIPPING: 500.00
                            </p>
                            <hr />
                            <p>
                              <b>
                                GRAND TOTAL: 6500.00
                              </b>
                            </p>
                          </div>
                        </Row>
                        <Link
                          className="btn"
                          to="/checkout"
                        >
                          <a
                            className="btn"
                            href="/checkout"
                            onClick={[Function]}
                          >
                            GO TO CHECKOUT
                          </a>
                        </Link>
                      </div>
                    </div>
                    <div
                      className="card-reveal"
                    >
                      <span
                        className="card-title"
                      >
                        Order Summary
                        <Icon
                          right={true}
                        >
                          <i
                            className="material-icons right"
                          >
                            close
                          </i>
                        </Icon>
                      </span>
                    </div>
                  </div>
                </Card>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </CartPage>
  </Router>
</BrowserRouter>
`;
